src/main/java/org/jahia/modules/sitemap/services/impl/SitemapServiceImpl.java

```diff
@@ -97,38 +94,28 @@ public Boolean sendSitemapXMLUrlPathToSearchEngines(String sitemapUrlPath) throw
         return true;
     }
 
-    public void flushSitemapEhCache() {
-        // TODO if on cluster we send a specific event to hazelcast otherwise we just flush current cache
+    @Override
+    public void flush() {
         logger.info("a flush of sitemap cache was triggered");
         sitemapCache.flush();
     }
 
-    public boolean isSitemapEhCacheEntryExist(String targetSitemapCacheKey) {
-        return sitemapCache.get(targetSitemapCacheKey) != null;
-    }
-
-    public String getSitemapEhCacheEntryValue(String targetSitemapCacheKey) {
-        return sitemapCache.get(targetSitemapCacheKey).getObjectValue().toString();
+    @Override
+    public String getSitemap(String key) {
+        return sitemapCache.get(key);
     }
 
-    public void addSitemapEhCacheEntry(String targetSitemapCacheKey, String sitemap, String sitemapCacheDuration) {
-
+    @Override
+    public void addSitemap(String key, String sitemap, String expiration) {
         // we get the desired cache expiration time in seconds
-        int expiredAt = getSitemapCacheExpirationInSeconds(sitemapCacheDuration);
-
-        Element sitemapCacheElement = new Element(targetSitemapCacheKey, sitemap);
-        sitemapCacheElement.setEternal(false);
-        sitemapCacheElement.setTimeToLive(expiredAt);
-
-        sitemapCache.put(sitemapCacheElement);
+        int expiredAt = getSitemapCacheExpirationInSeconds(expiration);
```

Comment:
Not necessary to declare a variable here you can replace by
 sitemapCache.put(key, sitemap, getSitemapCacheExpirationInSeconds(expiration));